### PR TITLE
feat: Add extra to sentry. [CONSVC-1886]

### PR DESCRIPTION
**Pull request by jrconlin**
* Update sentry to latest
* convert from chain-error to thiserror

### CODE REVIEW

File: tests/test_integration_all_rust.py [hunk 1]
```diff
@@ -690,6 +690,11 @@ def _ws_url(self):
     @max_logs(conn=4)
     def test_sentry_output(self):
         # Ensure bad data doesn't throw errors
+        ## The latest sentry library no longer uses the same
```
Comment (pjenvey) [inline tests/test_integration_all_rust.py:693]: I have a fix for this (capturing via the new sentry envelope API) here: https://github.com/mozilla-services/autopush-rs/compare/feat/1886-sentry...feat/1886-sentry-test

File: autopush/src/client.rs [hunk 2]
```diff
@@ -356,7 +358,11 @@ where
                     uaid.and_then(|uaid| Uuid::parse_str(uaid.as_str()).ok()),
                     Broadcast::from_hashmap(broadcasts.unwrap_or_default()),
                 ),
-                _ => return Err("Invalid message, must be hello".into()),
+                _ => {
+                    return Err(Error::BroadcastError(
```
Comment (pjenvey) [inline autopush/src/client.rs:362]: I think this is the only Error I'd call a "`BroadcastError`" in this module, the others here are more general client errors. Maybe just make those `GeneralError`s?

File: autopush/src/server/mod.rs [hunk 3]
```diff
@@ -390,12 +392,14 @@ impl Server {
                 handle.spawn(client.then(move |res| {
                     srv.open_connections.set(srv.open_connections.get() - 1);
                     if let Err(e) = res {
-                        let mut error = e.to_string();
+                        /*
```
Comment (pjenvey) [inline autopush/src/server/mod.rs:395]: You can kill this block, the `iter()` is a remnant from `error_chain`.

File: autopush-common/src/errors.rs [hunk 4]
```diff
@@ -29,80 +29,61 @@
 //! You can find some more documentation about this in the `error-chain` crate
 //! online.
 use std::any::Any;
-use std::error;
+//use std::error;
```
Comment (pjenvey) [inline autopush-common/src/errors.rs:32]:
```suggestion
```

File: autopush-common/src/errors.rs [hunk 5]
```diff
@@ -29,80 +29,61 @@
...
-        Io(io::Error);
-        Json(serde_json::Error);
-        Httparse(httparse::Error);
-        MetricError(cadence::MetricError);
-        UuidError(uuid::Error);
-        ParseIntError(num::ParseIntError);
-        ConfigError(config::ConfigError);
-    }
+#[derive(Error, Debug)]
+pub enum Error {
```
Comment (pjenvey) [inline autopush-common/src/errors.rs:40]: The big benefit of the separate `ErrorKind` enum and a `struct Error` is it lets us easily capture a `backtrace` on to a single field on the `Error` type.

Though autoconnect currently doesn't utilize the backtraces (in sentry) most of this common code should outlive its current impl., I think? So we should consider supporting backtraces out of the gate now.

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: autopush-common/src/db/commands.rs [hunk 6]
```diff
@@ -364,7 +370,8 @@ pub fn save_channels(
         },
         retryable_updateitem_error,
     )
-    .chain_err(|| "Error saving channels")
+    .map_err(|_e| ApcErrorKind::DatabaseError("Error saving channels".into()).into())
+    // .chain_err(|| "Error saving channels")
```
Comment (pjenvey) [inline autopush-common/src/db/commands.rs:374]:
```suggestion
```

File: autopush-common/src/db/commands.rs [hunk 7]
```diff
@@ -393,7 +400,8 @@ pub fn unregister_channel_id(
         move || ddb.update_item(update_item.clone()),
         retryable_updateitem_error,
     )
-    .chain_err(|| "Error unregistering channel")
+    .map_err(|_e| ApcErrorKind::DatabaseError("Error unregistering channel".into()).into())
+    //.chain_err(|| "Error unregistering channel")
```
Comment (pjenvey) [inline autopush-common/src/db/commands.rs:404]:
```suggestion
```

File: autopush-common/src/db/commands.rs [hunk 8]
```diff
@@ -447,7 +455,8 @@ pub fn lookup_user(
                     .send();
                 let response = drop_user(ddb, &uaid2, &router_table)
                     .and_then(|_| future::ok((hello_response, None)))
-                    .chain_err(|| "Unable to drop user");
+                    .map_err(|_e| ApcErrorKind::DatabaseError("Unable to drop user".into()).into());
+                //.chain_err(|| "Unable to drop user");
```
Comment (pjenvey) [inline autopush-common/src/db/commands.rs:459]:
```suggestion
```

File: autopush-common/src/lib.rs [hunk 9]
```diff
@@ -1,7 +1,6 @@
 #![recursion_limit = "1024"]
+#![allow(clippy::result_large_err)] // add this until refactor of `crate::error::ApcError result too large`
```
Comment (pjenvey) [inline autopush-common/src/lib.rs:2]: I think it usually just wants us to `Box` probably just the `Backtrace`

File: autopush/src/client.rs [hunk 10]
```diff
@@ -534,21 +549,34 @@ where
...
+                /*
+                Err(ApcErrorKind::Ws(_))
+                | Err(ApcErrorKind::Io(_))
+                | Err(ApcErrorKind::PongTimeout)
+                | Err(ApcErrorKind::RepeatUaidDisconnect)
+                | Err(ApcErrorKind::ExcessivePing)
+                | Err(ApcErrorKind::InvalidStateTransition(_, _))
+                | Err(ApcErrorKind::InvalidClientMessage(_))
+                | Err(ApcErrorKind::SendError) => None,
+                */
```
Comment (pjenvey) [inline autopush/src/client.rs:568]:
```suggestion
```

**Review by pjenvey — CHANGES_REQUESTED** [review]
Thanks for adding backtraces. There's a few more commented out lines of code, otherwise looks good.

File: autopush-common/src/lib.rs (hunk 9 above)
Comment (jrconlin) [reply autopush-common/src/lib.rs:2]: Tried that prior and clippy complained, loos like 1.67 might have fixed that?

**Review by pjenvey — APPROVED** [review]

**Pull request merged**